Added an example that uses the Moderation API to check for compliance in images

narenoai opened the pull request:
This example provides a way to use the moderation api to check if images are potentially harmful and if so why.

## Summary

This commit adds a python example to check an image using the moderation API. It also provides a way to batch check multiple images from URLs.

## Motivation

Using the moderation api to check if an image is complaint or not can be very useful for a lot of use cases. 

---

## For new content

When contributing new content, read through our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md), and mark the following action items as completed:

- [x] I have added a new entry in [registry.yaml](https://github.com/openai/openai-cookbook/blob/main/registry.yaml) (and, optionally, in [authors.yaml](https://github.com/openai/openai-cookbook/blob/main/authors.yaml)) so that my content renders on the cookbook website.
- [x] I have conducted a self-review of my content based on the [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md#rubric):
  - [x] Relevance: This content is related to building with OpenAI technologies and is useful to others.
  - [x] Uniqueness: I have searched for related examples in the OpenAI Cookbook, and verified that my content offers new insights or unique information compared to existing documentation.
  - [x] Spelling and Grammar: I have checked for spelling or grammatical mistakes.
  - [x] Clarity: I have done a final read-through and verified that my submission is well-organized and easy to understand.
  - [x] Correctness: The information I include is correct and all of my code executes successfully.
  - [x] Completeness: I have explained everything fully, including all necessary references and citations.

We will rate each of these areas on a scale from 1 to 4, and will only accept contributions that score 3 or higher on all areas. Refer to our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md) for more details.

phundal-openai commented:
Hi Naren,

Great work here... 2 items I would offer for consideration:

1. Borrow the layout, explanation, and dependency setup from this cookbook: https://cookbook.openai.com/examples/gpt4o/introduction_to_gpt4o

2. Consider taking this work and creating an additional image moderation section to this existing cookbook: https://github.com/openai/openai-cookbook/blob/17e9c2db6663cc772879580c4bcc2a5058f7f53b/examples/How_to_use_moderation.ipynb#L4

It may or may not make sense to build on the existing cookbook, but worth considering.

narenoai commented:
@phundal-openai I deleted the new cookbook and modified the existing moderation cookbook and added checking an image as an additional example. https://github.com/openai/openai-cookbook/pull/1653/commits/d0e5241c1b701bceaaa3c1b27eb75878a5133e48

phundal-openai reviewed (approved):
This looks great - nice addition.
